pkg/scalers/datadog_scaler.go

```diff
@@ -248,6 +253,11 @@ func validateClusterAgentMetadata(meta *datadogMetadata, config *scalersconfig.S
 	meta.HpaMetricName = "datadogmetric@" + meta.DatadogMetricNamespace + ":" + meta.DatadogMetricName
 	meta.DatadogMetricServiceURL = buildClusterAgentURL(meta.DatadogMetricsService, meta.DatadogNamespace, meta.DatadogMetricsServicePort)
 
+	// Set UseFiller flag if metricUnavailableValue is explicitly configured
+	if _, ok := config.TriggerMetadata["metricUnavailableValue"]; ok {
+		meta.UseFiller = true
+	}
```

Comment:
Isn't there any other way to detect if is has been defined than checking TriggerMetadata map? I'd prefer to not access it manually to avoid problems if the name changes or so. Maybe declaring `FillValue` as *float64 without default value and filling the default later as well as `UseFiller` does the same (I'm not sure tbh).